src/datarobotx/idp/batch_predictions.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2024 DataRobot, Inc. and its affiliates.
+#
+# All rights reserved.
+#
+# DataRobot, Inc.
+#
+# This is proprietary source code of DataRobot, Inc. and its
+# affiliates.
+#
+# Released under the terms of DataRobot Tool and Utility Agreement.
+# https://www.datarobot.com/wp-content/uploads/2021/07/DataRobot-Tool-and-Utility-Agreement.pdf
+
+from typing import Optional, TypedDict
+
+import datarobot as dr
+from datarobot.models.batch_job import IntakeSettings, OutputSettings, Schedule
+from datarobot.models.batch_prediction_job import BatchPredictionJobDefinition
+
+
+class JobSpec(TypedDict):
+    """Jobspec type definition.
+
+    num_concurrent : int (optional)
+            Number of concurrent chunks to score simultaneously. Defaults to
+            the available number of cores of the deployment. Lower it to leave
+            resources for real-time scoring.
+    intake_settings : dict (optional)
+            A dict configuring how data is coming from.
+
+            See supported options: https://github.com/datarobot/public_api_client/blob/master/datarobot/models/batch_prediction_job.py#L429
+
+    output_settings : dict (optional)
+            A dict configuring how scored data is to be saved.
+
+            See supported options: https://github.com/datarobot/public_api_client/blob/master/datarobot/models/batch_prediction_job.py#L474
+
+    """
+
+    num_concurrent: Optional[int]
+    deployment_id: str  # is this rquired
+    intake_settings: IntakeSettings
+    output_settings: Optional[OutputSettings]
+
+
+def get_update_or_create_batch_prediction_job(
```

Comment:
Namespace:
This function either updates a job or creates a job. `get` can be removed from the name